zanata-client-commands/src/main/java/org/zanata/client/commands/push/RawPushCommand.java

```diff
@@ -75,10 +96,174 @@ public RawPushCommand(PushOptions opts, RestClientFactory clientFactory) {
+     * input pattern: [extension;extension]
+     */
+    private List<String> extractExtensions(String rawExtensions) {
+        if(StringUtils.isEmpty(rawExtensions)) {
+            return Collections.emptyList();
+        }
+
+        //remove square bracket in first and last character
+        rawExtensions = rawExtensions.substring(1, rawExtensions.length()-1);
+        return Arrays.asList(rawExtensions.split(";"));
+    }
+
+    /**
+     * Extract input string to type and extensions
+     * input pattern: fileType[extension;extension], [extension;extension]
+     *
+     * @param typeWithExtension
+     * @return result[0] - type, result[1] - extensions. e.g [txt,html]
+     */
+    private String[] extractTypeAndExtension(String typeWithExtension) {
+        String[] result = new String[2];
+
+        final Pattern fileNameExtensionsPattern = Pattern.compile("(.*)?(\\[(.*?)\\])?");
+        Matcher matcher = fileNameExtensionsPattern.matcher(typeWithExtension);
+        if (matcher.find()) {
+            result[0] = matcher.group(1);
+            result[1] = matcher.group(2);
+        }
+        return result;
+    }
+
+    /**
+     * Validate inputFileType to make sure it's not empty
+     *
+     * @param inputFileType
+     * @param inputExtensions
+     * @param acceptedTypes
+     */
+    private void validateInputFileType(String inputFileType,
+            List<String> inputExtensions, List<DocumentType> acceptedTypes) {
+        if (StringUtils.isNotBlank(inputFileType)) {
+            return;
+        }
+        if (inputExtensions.isEmpty()) {
+            //throw error if inputFileType and inputExtensions is empty
+            throw new InvalidUserInputException(
+                    "Invalid expression for '--file-types' option");
+        } else {
+            //suggest --file-types options for this extension
+            for (DocumentType docType: acceptedTypes) {
+                for (String extension: docType.getSourceExtensions()) {
+                    if (inputExtensions.contains(extension)) {
+                        String msg =
+                                Messages.format(
+                                        "file.type.suggestFromExtension",
+                                        docType,
+                            extension, docType);
+                        throw new InvalidUserInputException(msg);
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Return map of validated type with set of extensions
+     *
+     * Validate user input file types against server accepted file types
+     *
+     * Valid input - properties_utf8,properties[txt],plain_text[md;markdown]
+     *
+     * - Each file type is only input once - e.g. html,properties,html
+     * - Same file extension cannot be in multiple file type - e.g. plain_text[txt],properties[txt]
+     * - Generate warning and ignore for source file extension that is not supported from file type. - e.g idml[txt]
```

Comment:
- [ ] we can make it more clear whether the examples are showing valid or invalid inputs, like this:

```
   * - Each file type is only input once. e.g.
   *    - valid: "html,properties,txt"
   *    - invalid: "html,properties,html"
```

- [ ] It looks like this is an example of a invalid input (due to use of `txt` twice). Better make it clear what the example is supposed to show.
- [ ] Even better, show an example of valid input, and an example of invalid input for each of the 3 rules above. The same examples can be copied to the help output and documentation for this option.